### PR TITLE
UCT/IB: enable ROCM direct rdma

### PR DESCRIPTION
**Pull request by yuq**

### CODE REVIEW

File: src/uct/ib/base/ib_md.c [hunk 1]
```diff
@@ -213,16 +213,31 @@ static ucs_status_t uct_ib_md_query(uct_md_h uct_md, uct_md_attr_t *md_attr)
...
+        /* check if ROCM KFD driver is loaded */
+        if (!access("/dev/kfd", F_OK)) {
+            md_attr->cap.reg_mem_types |= UCS_BIT(UCT_MD_MEM_TYPE_ROCM);
+            ucs_debug("%s: ROCM GPUDirect RDMA is enabled", uct_ib_device_name(&md->dev));
+            enabled = 1;
+        }
+
+        if (!enabled) {
+            if (md->config.enable_gpudirect_rdma == UCS_YES) {
+                ucs_error("%s: Couldn't enable GPUDirect RDMA. Please make sure"
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c]: i would add a function like check_gpudirect_driver(md, sys_file_name, mem_type, debug_name) to handle both GPU/ROCm and print the failure if could not find

Comment (yuq) [reply src/uct/ib/base/ib_md.c]: but this failure have to be when both ROCM and CUDA is not present, so can't be put in a signal function for each GPU.

Comment (yosefe) [reply src/uct/ib/base/ib_md.c]: i see, so this error message should still be common. only the ucs_debug() above should be inside the new function.
we can check here if  md_attr->cap.reg_mem_types has at least 1 non-host memtype

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9205/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9210/ for details (Mellanox internal link).

**Pull request merged**